src/Npgsql/TypeHandlers/TextHandler.cs

```diff
@@ -168,6 +168,47 @@ ValueTask<ArraySegment<char>> INpgsqlTypeHandler<ArraySegment<char>>.Read(Npgsql
             throw new NpgsqlSafeReadException(new NotSupportedException("Only writing ArraySegment<char> to PostgreSQL text is supported, no reading."));
         }
 
+        async ValueTask<byte[]> INpgsqlTypeHandler<byte[]>.Read(NpgsqlReadBuffer buf, int byteLen, bool async, FieldDescription fieldDescription)
+        {
+            var bytes = new byte[byteLen];
+            if (buf.ReadBytesLeft >= byteLen)
+            {
+                buf.ReadBytes(bytes, 0, byteLen);
+                return bytes;
+            }
+            else
+            {
+                if (byteLen <= buf.Size)
+                {
+                    // The bytes can fit in our read buffer, read it.
+                    while (buf.ReadBytesLeft < byteLen)
+                        await buf.ReadMore(async);
+                    buf.ReadBytes(bytes, 0, byteLen);
+                    return bytes;
+                }
+
+                // Bad case: the bytes don't fit in our buffer.
+                // This is rare - will only happen in CommandBehavior.Sequential mode (otherwise the
+                // entire row is in memory). Tweaking the buffer length via the connection string can
+                // help avoid this.
+
+                var pos = 0;
+                while (true)
+                {
+                    var len = Math.Min(buf.ReadBytesLeft, byteLen - pos);
+                    buf.ReadBytes(bytes, pos, len);
+                    pos += len;
+                    if (pos < byteLen)
+                    {
+                        await buf.ReadMore(async);
+                        continue;
+                    }
+                    break;
+                }
+                return bytes;
+            }
```

Comment:
Remove `else` and its brackets to decrease indent of the block. There is no reason to use it because you have `return` in the `if`.

```csharp
var bytes = new byte[byteLen]; 
if (buf.ReadBytesLeft >= byteLen) 
{ 
    buf.ReadBytes(bytes, 0, byteLen); 
    return bytes;
}

if (byteLen <= buf.Size)
// …
```

Reply:
Yeah, removing the else did make more sense in context. Thanks for the suggestion.